### PR TITLE
kernel: move priority queue handling to own file/header

### PR DESCRIPTION
**Pull request by nashif**
clean up headers under include/ and move handling of priority queue to
own file/header.
No need for the header  include/zephyr/kernel/internal/sched_priq.h
anymore. Move the relevant structures where they are being used in
kernel_structs.h.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by teburd — COMMENTED** [review]
I quite like the moves myself and think the cleanup is nice

File: kernel/include/priority_q.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ /* Multi Queue Scheduling */
+#elif defined(CONFIG_SCHED_MULTIQ)
+#define _priq_run_add		z_priq_mq_add
+#define _priq_run_remove	z_priq_mq_remove
+#define _priq_run_best		z_priq_mq_best
+static ALWAYS_INLINE void z_priq_mq_add(struct _priq_mq *pq, struct k_thread *thread);
+static ALWAYS_INLINE void z_priq_mq_remove(struct _priq_mq *pq, struct k_thread *thread);
+#endif
+
+/* Scalaable Wait Queue */
```
Comment (TaiJuWu) [inline kernel/include/priority_q.h:34]: Typo: scalable

**Review by TaiJuWu — APPROVED** [review]
Nice clean.

File: kernel/priority_queues.c [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	cmp = z_sched_prio_cmp(thread_a, thread_b);
+
+	if (cmp > 0) {
+		return true;
+	} else if (cmp < 0) {
+		return false;
+	} else {
+		return thread_a->base.order_key < thread_b->base.order_key
+			? 1 : 0;
+	}
```
Comment (pdgendt) [inline kernel/priority_queues.c:47]: A bit shorter?

```c
	if (cmp == 0) {
		return thread_a->base.order_key < thread_b->base.order_key;
	}

	return cmp > 0;
```

**Review by peter-mitsis — APPROVED** [review]

File: kernel/priority_queues.c (hunk 2 above)
Comment (nashif) [reply kernel/priority_queues.c:47]: sure, but I am just moving the code right now, improvments can be done on top.

**Review by dcpleung — APPROVED** [review]

**Pull request merged**